x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.tsx

```diff
@@ -0,0 +1,144 @@
+import { TOKEN_TYPE_DISPLAY_NAMES } from '../constants';
+import { apiTokenSort } from '../utils/api_token_sort';
+import { getModeDisplayText, getEnginesDisplayText } from '../utils';
+
+export const CredentialsList: React.FC = () => {
+  const { deleteApiKey, fetchCredentials, showCredentialsForm } = useActions(
+    CredentialsLogic
+  ) as ICredentialsLogicActions;
+
+  const { apiTokens, meta } = useValues(CredentialsLogic) as ICredentialsLogicValues;
+
+  const items = apiTokens.slice().sort(apiTokenSort);
+
+  const columns = [
+    {
+      name: 'Test',
+      width: '12%',
+      render: (item: IApiToken) => {
+        return <div>{item.name}</div>;
+      },
+    },
+    {
+      name: 'Type',
+      width: '18%',
+      render: (item: IApiToken) => {
+        return <div>{TOKEN_TYPE_DISPLAY_NAMES[item.type]}</div>;
+      },
+    },
+    {
+      name: 'Key',
+      width: '43%',
+      render: (item: IApiToken) => {
+        return (
+          <div>
+            <EuiCopy
+              textToCopy={item.key || ''}
+              afterMessage={i18n.translate('xpack.enterpriseSearch.appSearch.credentials.copied', {
+                defaultMessage: 'Copied',
+              })}
+            >
+              {(copy) => (
+                <div>
+                  <EuiButtonIcon
+                    onClick={copy}
+                    iconType="copyClipboard"
+                    aria-label={i18n.translate(
+                      'xpack.enterpriseSearch.appSearch.credentials.copyApiKey',
+                      {
+                        defaultMessage: 'Copy API Key to clipboard',
+                      }
+                    )}
+                  />
+                  {item.key}
+                </div>
+              )}
+            </EuiCopy>
+          </div>
+        );
+      },
+    },
+    {
+      name: 'Modes',
+      width: '10%',
+      render: (item: IApiToken) => {
+        return <div>{getModeDisplayText(item)}</div>;
+      },
+    },
+    {
+      name: 'Engines',
+      width: '10%',
+      render: (item: IApiToken) => {
+        return <div>{getEnginesDisplayText(item)}</div>;
+      },
+    },
+    {
+      width: '4%',
+      render: (item: IApiToken) => {
+        return (
+          <EuiButtonIcon
+            onClick={() => showCredentialsForm(item)}
+            iconType="pencil"
+            aria-label={i18n.translate('xpack.enterpriseSearch.appSearch.credentials.editKey', {
+              defaultMessage: 'Edit API Key',
+            })}
+          />
+        );
+      },
+    },
+    {
+      width: '4%',
+      render: (item: IApiToken) => {
+        return (
+          <EuiButtonIcon
+            onClick={() => item.name && deleteApiKey(item.name)}
+            iconType="trash"
+            aria-label={i18n.translate('xpack.enterpriseSearch.appSearch.credentials.deleteKey', {
+              defaultMessage: 'Delete API Key',
+            })}
+          />
+        );
+      },
+    },
```

Comment:
Super exciting proposal here, I think you'll like it - can we use EUI's native `actions` column? Ref: https://elastic.github.io/eui/#/tabular-content/tables#adding-actions-to-table

Here's what the code looks like: 
```suggestion
    {
      actions: [
        {
          name: i18n.translate('xpack.enterpriseSearch.actions.edit', {
            defaultMessage: 'Edit',
          }),
          description: i18n.translate('xpack.enterpriseSearch.appSearch.credentials.editKey', {
            defaultMessage: 'Edit API Key',
          }),
          type: 'icon',
          icon: 'pencil',
          color: 'primary',
          onClick: (item: IApiToken) => showCredentialsForm(item),
        },
        {
          name: i18n.translate('xpack.enterpriseSearch.actions.delete', {
            defaultMessage: 'Delete',
          }),
          description: i18n.translate('xpack.enterpriseSearch.appSearch.credentials.deleteKey', {
            defaultMessage: 'Delete API Key',
          }),
          type: 'icon',
          icon: 'trash',
          color: 'danger',
          onClick: (item: IApiToken) => item.name && deleteApiKey(item.name),
        },
      ],
    },
```

Here's what we get from it:

- Handy-dandy UI tooltips (presumably screen-reader accessible because EUI rocks):
  <img width="1007" alt="" src="https://user-images.githubusercontent.com/549407/95272953-9d756d80-07f6-11eb-988c-936291dd957f.png">
- Better mobile UX
  <img width="643" alt="" src="https://user-images.githubusercontent.com/549407/95272947-9a7a7d00-07f6-11eb-8a66-f5ded2fa4b35.png">
  (For context: I didn't use an actions column on the Engines Overview table because that one uses text ("Manage") instead of an icon, and that looked super weird on mobile)

All in all, I think we gain a lot in UX by switching - LMK what you think!

FYI, to make this change, we also need to add an extra type def on line 33 where we declare `columns` -

```js
  const columns: Array<EuiBasicTableColumn<IApiToken>> = [
```

I had the same type struggles when working w/ the Engines tables so I think the columns definition is something to keep an eye out for whenever we use EuiBasicTable in the future.

Reply:
Holy moly, I didn't realize that existed. v cool, ty.

Reply:
53865ce